### PR TITLE
Use new FluxC code to trigger close account API request and handle errors

### PR DESCRIPTION
**Pull request by JorgeMucientes**
Part of: #8955. And continuation of [PR](https://github.com/woocommerce/woocommerce-android/pull/9033)
This PR relies on these [FluxC](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2732) changes so please don't merge until those are merged and until the base branch for this PR is `trunk`

### Description

Makes use of the new API endpoint to close account 

### Testing instructions

For now, just navigate to settings and attempt to close the account. You'll see an error because you can't close the account if you have active stores. 
**Note**: You must be logged in with a WP.com account. Application passwords login won't display this new "Close account" setting. 

### Images/gif

https://github.com/woocommerce/woocommerce-android/assets/2663464/e9f9daf6-58a3-4f1e-92c4-ed98dd84d8c8

### CODE REVIEW

File: WooCommerce/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -1900,7 +1900,8 @@
     <string name="settings_close_account_dialog_confirm_button">Permanently Close Account</string>
     <string name="settings_close_account_dialog_loading_title">Closing account…</string>
     <string name="settings_close_account_error_dialog_title">Couldn\'t close account</string>
-    <string name="settings_close_account_error_dialog_description">An error occurred while attempting to close your account.</string>
+    <string name="settings_close_account_generic_error_description">An error occurred while attempting to close your account.</string>
+    <string name="settings_close_account_active_stores_error_description">This account cannot be close while it has active stores.</string>
```
Comment (0nko) [inline WooCommerce/src/main/res/values/strings.xml]: NP: Spelling
```suggestion
    <string name="settings_close_account_active_stores_error_description">This account cannot be closed while it has active stores.</string>
```

**Review by 0nko — COMMENTED** [review]
>  You'll see an error because you can't close the account if you have active stores.

When I attempt to close the account I get a generic error, not the active stores error. Is that expected?

image

**Comment by JorgeMucientes** [comment]
Thanks for the review @0nko. 
In this PR it is not expected to show the `account has active stores` error. That error handling was added in the next [PR](https://github.com/woocommerce/woocommerce-android/pull/9050). I've just tested again running the app from branch `feature/8955-close-account-from-site-picker` and closing the account and it displayed the expected error (with the typo you noticed though, but that will be fixed once we merge this PR and the other is rebased to `trunk`): 

Screenshot 2023-05-18 at 18 45 06

**Comment by JorgeMucientes** [comment]
The PR can't be merged though, until the corresponding [FluxC](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2732) changes are merged and I update the `build.gradle` file with the correct fluxC change set. That's the reason checks are failing now after I updated the PR with `trunk` changes to resolve the conflicts

**Review by 0nko — APPROVED** [review]
@JorgeMucientes I've merged the FluxC PR. You can go ahead and update the hash.

**Pull request merged**